### ISSUE TITLE
About different random masks for each sample in a batch

### ISSUE THREAD

**Issue by ZrrSkywalker** [LayneH/GreenMIM#2]
Thanks for the great work!
I notice that GreenMIN requires the same mask in a batch for training efficiency. I wonder how much the sample-wise mask would affect the performance. Could you please give some quantitive results?
Many thanks! I would be highly appreciated.

**Comment by LayneH** [comment]
Hi,

This is indeed a good question.
Because there are a lot of nuisances to making GreenMIM work with sample-wise masks (e.g., handling the arbitrary group sizes within a batch), I did not perform a rigorous comparison of these two masking strategies.

I am now planning to compare the performance of SimMIM with sample-/batch-wise masks to inspect the influence of the mask strategies. I will get you updated when I have the results.

**Comment by LayneH** [comment]
Hi,

Sorry for the late reply.
The results of the aforementioned experiments are as below,
| Method | Mask Strategy | IN1K Acc. |
| -------  | --------------- | --------- |
| SimMIM| Sample-wise    | 83.38%    |
| SimMIM| Batch-wise       | 83.31%     |

*Note that we use a batch size of 1024 at the fine-tuning stage for both models.

The results suggest that the masking strategy does not affect the performance too much.
After all, the positions of objects vary in each image so that there remains enough randomness in the masking process.

**Comment by ZrrSkywalker** [comment]
Thanks for your experiments and explanation!
I see. The variation of objects in images can indeed provide enough randomness that eliminates the batch-wise masking.
